pkg/minishift/addon/manager/addon_manager.go

```diff
@@ -196,6 +217,33 @@ func (m *AddOnManager) ApplyAddOn(addOn addon.AddOn, context *command.ExecutionC
 	return nil
 }
 
+func (m *AddOnManager) RemoveAddOn(addOn addon.AddOn, context *command.ExecutionContext) error {
+	fmt.Print(fmt.Sprintf("-- Removing addon '%s':", addOn.MetaData().Name()))
+	context.AddToContext("addon-name", addOn.MetaData().Name())
+	defer context.RemoveFromContext("addon-name")
+
+	err := m.verifyRequiredVariablesInContext(context, addOn.MetaData())
+	if err != nil {
+		return err
+	}
+
+	oldDir, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "Unable to remove addon due to failing IO operation")
+	}
+	defer os.Chdir(oldDir)
+
+	os.Chdir(addOn.InstallPath())
+	for _, c := range addOn.Commands() {
```

Comment:
This ties into the design discussion. You pass the add-on and then iterate the add-on commands. IMO this is confusing. It would be nicer here to pass an add-on, but then since it is the remove function, iterate the RemoveCommands.